site/src/pages/RoadmapPage/ImportZot4PlanPopup.tsx

```diff
@@ -0,0 +1,157 @@
+import { FC, useContext, useState } from 'react';
+import './ImportZot4PlanPopup.scss';
+import { FileEarmarkArrowDown, ExclamationTriangle } from 'react-bootstrap-icons';
+import { Button, Form, Modal } from 'react-bootstrap';
+import { addRoadmapPlan, setPlanIndex, selectAllPlans } from '../../store/slices/roadmapSlice';
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import ThemeContext from '../../style/theme-context';
+import trpc from '../../trpc.ts';
+import { expandAllPlanners } from '../../helpers/planner';
+import spawnToast from '../../helpers/toastify';
+import helpImage from '../../asset/zot4plan-import-help.png';
+
+const ImportZot4PlanPopup: FC = () => {
+  const dispatch = useAppDispatch();
+  const { darkMode } = useContext(ThemeContext);
+  const [showModal, setShowModal] = useState(false);
+  const [scheduleName, setScheduleName] = useState('');
+  const [studentYear, setStudentYear] = useState('1');
+  const [busy, setBusy] = useState(false);
+  const allPlanData = useAppSelector(selectAllPlans);
+
+  const obtainImportedRoadmap = async (schedName: string, currYear: string) => {
+    // Get the result
+    const result = await trpc.zot4PlanImportRouter.getScheduleFormatted.query({
+      scheduleName: schedName,
+      studentYear: currYear,
+    });
+    // Verify that the result has one planner (if not, the import failed)
+    if (result.planners.length == 0) {
+      // Notify the user
+      spawnToast('The schedule "' + schedName + '" could not be successfully retrieved', true);
+      return;
+    }
+    // Expand the result
+    const expandedPlanners = await expandAllPlanners(result.planners);
+    // Check for validity: length and invalid course names
+    if (expandedPlanners.length < 1) {
+      spawnToast('The schedule "' + schedName + '" could not be successfully imported', true);
+      return;
+    }
+    let problemCount = 0;
+    for (const yearPlan of expandedPlanners[0].content.yearPlans) {
+      for (const quarter of yearPlan.quarters) {
+        let i = 0;
+        while (i < quarter.courses.length) {
+          if (quarter.courses[i] === undefined) {
+            // An unknown course name was encountered when expanding the planner;
+            // It will crash PeterPortal if loaded, so remove it
+            quarter.courses.splice(i, 1);
+            problemCount++;
+          } else {
+            i++;
+          }
+        }
+      }
+    }
+    if (problemCount > 0) {
+      spawnToast(
+        'The schedule "' + schedName + '" contained ' + problemCount + ' unknown course(s), which were removed',
```

Comment:
This message is a bit long for a toast, maybe something more like `Partially imported "${schedName}" (removed ${problemCount} unknown course(s))`